webui/src/pages/RequestsViewer.tsx

```diff
@@ -0,0 +1,33 @@
+import Container from "@mui/material/Container";
+import { useCallback, useEffect, useState } from "react";
+import RequestCard from "../components/RequestCard";
+import { fetchRequests } from "../rpc/backend";
+import { RequestsResponse } from "../rpc/models";
+
+const RequestsViewer = () => {
+  const [notFetched, setNotFetched] = useState(true);
+  const [requests, setRequests] = useState<RequestsResponse>();
+
+  const doRequestsFetch = useCallback(() => {
+    fetchRequests(
+      res => setRequests(res),
+      err => console.error(err),
+    ).then(() => setNotFetched(false));
+  }, []);
+
+  useEffect(() => {
+    if (notFetched) {
+      doRequestsFetch();
+    }
+  }, [doRequestsFetch, notFetched]);
```

Comment:
not as part of this PR, but if you get bored at some point then react router has some way of doing this automatically. if you want to look at it. it might not work well with typescript or just be good